roles/uki/tasks/main.yml

```
@@ -0,0 +1,307 @@
+        state: directory
+      register: mok_dir
+      changed_when: false
+
+    - name: Extract the certificate
+      become: true
+      ansible.builtin.shell:
+        cmd: >
+          set -o pipefail && certutil
+          -d '{{ uki_mok.database_path }}'
+          -n '{{ uki_mok.friendly_name }}'
+          -Lr | base64
+        executable: /bin/sh
+      register: mok_export
+      changed_when: false
+
+    - name: Copy the certificate to the temporary directory
+      become: true
+      ansible.builtin.copy:
+        content: "{{ mok_export.stdout | b64decode }}"
+        dest: "{{ mok_dir.path }}/MOK.der"
+        mode: '0644'
+      changed_when: false
+
+    - name: Check its enrollment status
+      become: true
+      ansible.builtin.command:
+        argv:
+          - mokutil
+          - -t
+          - "{{ mok_dir.path }}/MOK.der"
+      register: mok_test
+      changed_when: false
+      failed_when:
+        - mok_test.rc != 1
+        - '"already enrolled" not in mok_test.stdout'
+
+- name: Get a temporary directory
+  ansible.builtin.tempfile:
+    suffix: crichez.secureboot.uki
+    state: directory
+  delegate_to: localhost
+  register: clone_dir
+  changed_when: false
+
+- name: Clone virt-firwmare
+  ansible.builtin.git:
+    repo: 'https://gitlab.com/kraxel/virt-firmware'
+    dest: "{{ clone_dir.path }}"
+    version: v24.11
+  delegate_to: localhost
+  changed_when: false
+
+- name: Set configuration and build UKI
+  vars:
+    install_conf_path: /etc/kernel/install.conf
+    uki_conf_path: /etc/kernel/uki.conf
+    install_script_path: /etc/kernel/install.d/99-uki-uefi-setup.install
+    boot_svc_name: kernel-bootcfg-boot-successful.service
+    boot_svc_path: /etc/systemd/system/{{ boot_svc_name }}
+  become: true
+  block:
+    - name: Back up install.conf
+      ansible.builtin.slurp:
+        src: "{{ install_conf_path }}"
+      register: install_conf_backup
+      changed_when: false
+      ignore_errors: true
```

We should not be ignoring errors on backup tasks. If the file doesn't exist, we should skip the task. If the file exists and the backup fails, we should restore any changes and fail.